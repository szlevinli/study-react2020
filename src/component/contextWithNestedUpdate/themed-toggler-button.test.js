import { render } from '@testing-library/react';
import React from 'react';
import { ThemeContext, themes } from './theme-context';
import ThemedTogglerButton from './themed-toggler-button';
import userEvent from '@testing-library/user-event';

// eslint-disable-next-line react/prop-types
const MockComponent = ({ contextValue }) => {
  return (
    <ThemeContext.Provider value={contextValue}>
      <ThemedTogglerButton></ThemedTogglerButton>
    </ThemeContext.Provider>
  );
};

it('should be render correctly', () => {
  const { container } = render(
    <MockComponent
      contextValue={{
        theme: themes.light,
        toggleTheme: jest.fn(),
      }}
    ></MockComponent>
  );
  expect(container).toMatchSnapshot();
});

it('should be render a button', () => {
  const { getByRole } = render(
    <MockComponent
      contextValue={{
        theme: themes.light,
        toggleTheme: jest.fn(),
      }}
    ></MockComponent>
  );
  expect(getByRole('button')).toHaveTextContent('Toggle Theme');
  expect(getByRole('button')).toHaveAttribute(
    'style',
    expect.stringContaining('background-color')
  );
});

it('should be handle click', () => {
  const mockFn = jest.fn();
  const { getByRole } = render(
    <MockComponent
      contextValue={{
        theme: themes.light,
        toggleTheme: mockFn,
      }}
    ></MockComponent>
  );
  userEvent.click(getByRole('button'));
  expect(mockFn).toHaveBeenCalledTimes(1);
});
