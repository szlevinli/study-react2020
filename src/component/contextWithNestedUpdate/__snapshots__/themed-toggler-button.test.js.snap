// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be render correctly 1`] = `
<div>
  <button
    style="background-color: rgb(238, 238, 238);"
  >
    Toggle Theme
  </button>
</div>
`;
